.claude/hooks/gdd-allowlist-bridge.sh

```
@@ -0,0 +1,457 @@
+# remaining Tier 1 checks: a command like `grep foo file > out`
+# matched the grep arm, the inner case found no `|`, and the outer
+# case completed silently — bypassing the redirect deny. By checking
+# the more-specific deniable operators first, the grep arm only
+# fires when the ONLY Tier 1 violation is `|` (regex alternation or
+# a grep-pipeline), which is the case where "use the Grep tool" is
+# the right corrective message.
+case "$cmd" in
+    *"&&"*|*"||"*|*";"*)
+        deny "Shell composition (\&\&, ||, ;) is disallowed by this hook. Run each command as a separate tool call so the harness can validate each segment independently. If you need conditional behavior, check the result of one call before issuing the next."
+        ;;
+    *'`'*|*'$('*)
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)